packages/office-ui-fabric-react/src/components/Keytip/Keytip.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { BaseComponent, IPoint, getDocument, mergeOverflowKeySequences } from '../../Utilities';
+import { Callout } from '../../Callout';
+import { DirectionalHint } from '../../ContextualMenu';
+import { IKeytip, IKeytipProps } from './Keytip.types';
+import { KeytipContent } from './KeytipContent';
+import { getCalloutStyles, getCalloutOffsetStyles } from './Keytip.styles';
+import { constructKeytipTargetFromSequences } from '../../utilities/keytips';
+
+/**
+ * A callout corresponding to another Fabric component to describe a key sequence that will activate that component
+ *
+ * @export
+ * @class Keytip
+ * @extends {BaseComponent<IKeytipProps, {}}>}
+ */
+export class Keytip extends BaseComponent<IKeytipProps, {}> implements IKeytip {
+  // tslint:disable-next-line:no-any
+  constructor(props: IKeytipProps, context: any) {
+    super(props, context);
+  }
+
+  public render(): JSX.Element {
+    const {
+      keySequences,
+      offset,
+      overflowSetSequence
+    } = this.props;
+    let {
+      calloutProps
+    } = this.props;
+
+    let keytipTarget: string;
+    // Take into consideration the overflow sequence
+    if (overflowSetSequence) {
+      keytipTarget = constructKeytipTargetFromSequences(mergeOverflowKeySequences(keySequences, overflowSetSequence));
+    } else {
+      keytipTarget = constructKeytipTargetFromSequences(keySequences);
+    }
+
+    if (offset) {
+      // Set callout to top-left corner, will be further positioned in
+      // getCalloutOffsetStyles
+      calloutProps = {
+        ...calloutProps,
+        coverTarget: true,
+        directionalHint: DirectionalHint.topLeftEdge
+      };
+    }
+
+    if (!calloutProps || calloutProps.directionalHint === undefined) {
```

Comment:
Yes it can be undefined, but spreading an undefined object is ok. Only throws an error when trying to do it with arrays. See: https://stackoverflow.com/questions/47155141/spreading-undefined-in-array-vs-object